## 🤖 Re-review: feat(dev): document OM_WATCH_SCOPE dev-memory lever + emoji watch-mode log (auto-review-pr)

**Verdict: CHANGES REQUESTED** — one High-severity finding: the PR breaks a pre-existing unit test, and the required `test` CI check is **red** on the current head (`e1d6de616`).

> Note on prior state: this PR carried a comment-only "APPROVED" from an earlier pass and was routed to `merge-queue`. That approval was posted as a PR comment, not a formal GitHub review (`reviewCount: 0`), and **two commits landed after it** (`0c528c139`, `e1d6de616`). The current head was never actually reviewed, and it is failing CI — so `merge-queue` was stale. Moving it back to `changes-requested`.

### 🔴 High — emoji watch-scope log breaks existing test `watch-packages-scope.test.mjs`

The emoji feature changed the watch-scope log format:

`scripts/watch-packages.mjs:235`
```diff
-    log(`[watch] watch scope: ${selection.mode} — ${selection.reason}`)
+  const watchMode = describeWatchMode(selection.mode)
+  log(`[watch] watch scope: ${watchMode.emoji} ${watchMode.mode} — ${selection.reason}`)
```

The pre-existing test `scripts/__tests__/watch-packages-scope.test.mjs:47-50` (present on `develop`, **not** touched by this PR) asserts on the old format:

```js
assert.ok(
  logs.some((line) => line.includes('watch scope: env')),
  `expected watch scope log, got: ${logs.join('\n')}`,
)
```

With the emoji inserted, the log is now `[watch] watch scope: 🎯 env — watching 2 package(s) from explicit selection`, so `'watch scope: env'` is no longer a substring (the emoji + space sits between `watch scope: ` and `env`). The assertion fails and the `test` job exits 1.

CI evidence (run `28871978053`, job `test`, head `e1d6de616`):
```
test at scripts/__tests__/watch-packages-scope.test.mjs:26:1
✖ runConsolidatedWatch honors OM_WATCH_SCOPE=env and tracks only the listed packages
  AssertionError [ERR_ASSERTION]: expected watch scope log, got:
    [watch] watch scope: 🎯 env — watching 2 package(s) from explicit selection
  ...
ℹ tests 315  ℹ pass 314  ℹ fail 1
##[error]Process completed with exit code 1.
```

**Required fix:** update the assertion in `scripts/__tests__/watch-packages-scope.test.mjs:48` to match the new emoji-decorated format — e.g. assert on `line.includes('env — watching')` or `/watch scope: .* env /`, matching the same style already used in the new `scripts/__tests__/watch-scope.test.mjs`. The PR added tests for the `describeWatchMode` helper (`watch-scope.test.mjs`) but missed updating this dependent integration-style test that exercises the real log line.

### ✅ Otherwise clean

- **Docs / env changes** (`README.md`, `apps/docs/docs/appendix/troubleshooting.mdx`, `apps/docs/docs/installation/monorepo.mdx`, `apps/mercato/.env.example`, `packages/create-app/template/.env.example`) — documenting `OM_WATCH_SCOPE`: consistent, accurate, no contract-surface impact.
- **`describeWatchMode` helper** (`scripts/watch-scope.mjs`) + its unit test (`scripts/__tests__/watch-scope.test.mjs`) — well-scoped, covers every known scope and the fallback.
- **`AdvancedFilterPanel.tsx`** — the "stray `ded` token" removal (`0c528c139`) nets out clean against `develop` (the token was self-introduced earlier in the branch and removed here; it is not present on `develop`). No net change to that file relative to base. Out-of-scope for the PR's stated purpose but harmless.
- No tenant-scoping, encryption, event-ID, DB-schema, or other frozen-contract-surface concerns. Dev-tooling only.

### Labels
Removing `merge-queue`, applying `changes-requested`. Keeping `documentation`, `feature`, `skip-qa`, `priority-low`, `risk-low` (blast radius is isolated dev-watch tooling; skip-qa remains correct — no customer-facing surface).

### Summary
| Severity | Count |
|----------|-------|
| Critical | 0 |
| High     | 1 |
| Medium   | 0 |
| Low      | 0 |

Single blocker: fix the broken assertion in `scripts/__tests__/watch-packages-scope.test.mjs` so the `test` CI check goes green. Per instructions this run is review-only (no autofix applied).

--- a/scripts/__tests__/watch-packages-scope.test.mjs
+++ b/scripts/__tests__/watch-packages-scope.test.mjs
@@ -45,7 +45,7 @@ test('runConsolidatedWatch honors OM_WATCH_SCOPE=env and tracks only the listed 
 
     assert.deepEqual(packages.map((p) => p.shortLabel).sort(), ['alpha', 'charlie'])
     assert.ok(
-      logs.some((line) => line.includes('watch scope: env')),
+      logs.some((line) => /watch scope: \S+ env\b/.test(line)),
       `expected watch scope log, got: ${logs.join('\n')}`,
     )
   } finally {

